v3/mux.go

```diff
@@ -229,41 +275,57 @@ func (m *Mux) readLoop() {
 			m.setErr(fmt.Errorf("peer sent invalid frame ID (%v) (covert=%v, length=%v, flags=%v)", h.id, covert, h.length, h.flags))
 			return
 		}
+
 		// look for matching Stream
-		if curStream == nil || h.id != curStream.id {
-			m.mu.Lock()
-			if s := m.streams[h.id]; s != nil {
-				curStream = s
-			} else {
-				if h.flags&flagFirst == 0 {
-					// we don't recognize the frame's ID, but it's not the
-					// first frame of a new stream either; we must have
-					// already closed the stream this frame belongs to, so
-					// ignore it
+		var stream *Stream
+		m.mu.Lock()
```

Comment:
why was the `curStream` optimization removed? (tbf, I don't think I ever specifically benchmarked it, so maybe the benefit was negligible -- but I did think it was cute)